test/e2e_node/image_list.go

```diff
@@ -60,12 +60,19 @@ var NodePrePullImageList = sets.NewString(
 	imageutils.GetE2EImage(imageutils.Perl),
 	imageutils.GetE2EImage(imageutils.Nonewprivs),
 	imageutils.GetPauseImageName(),
-	getGPUDevicePluginImage(),
 	imageutils.GetE2EImage(imageutils.NodePerfNpbEp),
 	imageutils.GetE2EImage(imageutils.NodePerfNpbIs),
 	imageutils.GetE2EImage(imageutils.NodePerfTfWideDeep),
 )
 
+func init() {
```

Comment:
Sticking this in `init` doesn't seem right when there's nothing else in here... is there somewhere else we can put this logic? Why is this only an issue for the `getGPUDevicePluginImage` and `getSRIOVDevicePluginImage`?

Reply:
`getGPUDevicePluginImage` and `getSRIOVDevicePluginImage` in a bad network, may return empty string, causes subsequent pull image faill.or delete empty strings when using?

Reply:
Sure but there's also `getNodeProblemDetectorImage` which you're not changing. It just seems weird to me that we're managing this image in an `init()` and the other two images in a code block. If we are dynamically setting some images I think they should all be grouped together, as is this is very hard to follow.

Reply:
If we follow the pattern from the other calls then this has to insert _some_ value into the list, we could do that and put back `.Delete("")` on the end, but otherwise it's going to look fairly similar to this to accommodate maybe inserting maybe not?

IMHO this list should not be a global and should be initialized during main, so e.g. logging can be initialized first, but that's a larger change.